### ISSUE TITLE
About Reparameterization Trick of spike variable "gamma"

### ISSUE THREAD

**Issue by Ronosu** [Alfo5123/Variational-Sparse-Coding#2]
You said in VSC paper that 
log(gamma) = - ReLu (-v_out) ・・・(1) , 
but if the activation function in hidden layer is "sigmoid" or "ReLu", 
I think that (1) out the 0s in all dimensions.
So the model can't carry  any information.
(But your experiments show  good results!) 

Would you teach me why this reparameterization trick works?
(I think log(gamma) = - ReLu (+v_out) will work well than (1))

In addition, how did you decide the range c-annealing from 50 to 200?

**Comment by Alfo5123** [comment]
Hi @Ronosu thanks for your comment. 

Indeed,  formula 12 of the paper (https://openreview.net/pdf?id=SkeJ6iR9Km) shows what you mentioned. As you can see in the code from 'vsc.py' we write, '-F.relu(-self.fc23(h1))'. That is, we pass the output through a fully connected layer before changing the sign and passing through activation. This causes the value to not be zero, and carry information.  

Regarding the second question, it was not our decision, it was suggested by the authors of the paper as you can see in the section C2 of the appendix, in the link above.
